src/scraper/scraper.cpp

```diff
@@ -3001,14 +3048,38 @@ ScraperStats GetScraperStatsByConsensusBeaconList()
     return mScraperStats;
 }
 
+unsigned int GetActiveProjectCountFromConvergedManifest(const ConvergedManifest& StructConvergedManifest)
+{
+    unsigned int nActiveProjects = 0;
+
+    const NN::WhitelistSnapshot projectWhitelist = NN::GetWhitelist().Snapshot();
+
+    const NN::GreylistSnapshot greylist = NN::Quorum::FilterGreylist(projectWhitelist);
+
+    // Note we cannot just use greylist.ActiveProjects().size() here, because the Manifest may be missing a
+    // project that is on the whitelist due to lack of availability.
+    for (const auto& entry : StructConvergedManifest.ConvergedManifestPartsMap)
+    {
+        if (projectWhitelist.Contains(entry.first) && !greylist.Contains(entry.first)) ++nActiveProjects;
```

Comment:
Here we want to use:

```c++
&& greylist.CheckActive(entry.first)
```

Reply:
Yep

Reply:
And we don't need the Whitelist.Contains here.... it is superfluous.